### PR TITLE
Add pre-commit with autopep8 + change made by it

### PR DESCRIPTION
**Pull request by fruch**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# See https://pre-commit.com for more information
```
Comment (bentsi) [inline .pre-commit-config.yaml]: can you add documentation to Readme, of how to use and install it?

Comment (amoskong) [reply .pre-commit-config.yaml]: It's very cool and smart in my practice.
```
sudo pip install pre-commit
pre-commit
<update file>
git commit -sa 
<automatically check and fix>
```

We need to document that how to disable it.

Comment (fruch) [reply .pre-commit-config.yaml]: `pre-commit uninstall` to remove the hook
and 
git commit ... -n to disable the hooks

Comment (fruch) [reply .pre-commit-config.yaml]: I'll update the readme to explain those a bit

**Comment by fruch** [comment]
@bentsi I think this one is safe enough to merge, next step would be pylint...

**Comment by amoskong** [comment]
Hi @fruch can you also add pre-commit to dtest? which has more style problem.

**Pull request merged**